### PR TITLE
Adds support for the shoulda-context gem

### PR DESCRIPTION
**Pull request by alexpjohnson**
From #32

### CODE REVIEW

File: lib/knapsack/adapters/minitest_adapter.rb [hunk 1]
```diff
@@ -48,7 +48,7 @@ def set_test_helper_path(file_path)
 
       def self.test_path(obj)
         # Pick the first public method in the class itself, that starts with "test_"
-        test_method_name = obj.public_methods(false).select{|m| m =~ /^test_/ }.first
```
Comment (ArturT) [inline lib/knapsack/adapters/minitest_adapter.rb]: When you use the previous version of the code with `/^test_/` then test for shoulda-context still passes. It should fail.

**Comment by ArturT** [comment]
@alexpjohnson  I had to revert this PR because it doesn't work and test is false positive. 

I have here a separate rails app. I added there shoulda-context and I tried to generate report to see if the test file path is detected correctly but it is not.
You can find here more how to test it https://github.com/KnapsackPro/rails-app-with-knapsack/pull/1

Basically what we would like to achieave here is to have correct test file path in the variable `full_test_path`. Maybe you will find how to do it. I was trying for a while but I couldn't find a way to get test file path based on `obj`.

```ruby
      def self.test_path(obj)
        # Pick the first public method in the class itself, that starts with "test_"
        test_method_name = obj.public_methods(false).select{|m| m =~ /^test(:|_)/ }.first
        method_object = obj.method(test_method_name)
        full_test_path = method_object.source_location.first
        parent_of_test_dir_regexp = Regexp.new("^#{@@parent_of_test_dir}")
        test_path = full_test_path.gsub(parent_of_test_dir_regexp, '.')
        # test_path will look like ./test/dir/unit_test.rb
        test_path
      end

```

**Comment by alexpjohnson** [comment]
Hey @ArturT, I've done a bit more investigating and it seems the root issue is that shoulda-context is breaking the API established by minitest. By dynamically generating methods, they are losing the original `source_location` of the test. I'm still tinkering around with it, but as that gem is currently designed I don't think it's possible for you to fix it in this gem, nor do I think it's really your responsibility. I've opened up an issue on their gem [here](https://github.com/thoughtbot/shoulda-context/issues/50). I'll keep tinkering around with it, but hopefully an upstream PR will get merged and then your code should just work :) Thanks for your help, and sorry for the premature PR.

**Comment by ArturT** [comment]
@alexpjohnson Thanks for the update. Hope shoulda-context team will have some tips how to determine the path of the test file. Maybe there is another way than using `source_location`.

**Comment by alexpjohnson** [comment]
Hey @ArturT I've got a solution that works locally for me. I refactored a bit of the shoulda-context gem to allow passing in the original source location. There's a PR open [here](https://github.com/thoughtbot/shoulda-context/pull/51).

**Comment by ArturT** [comment]
@alexpjohnson With your fix in shoulda-context there is no need to do changes in knapsack as all methods generated by shoulda-context have prefix `test_`, right?

**Comment by alexpjohnson** [comment]
Correct. It should all just happen auto-magically :)

**Comment by alexpjohnson** [comment]
Just wanted to keep you up to date in case any of your other users mentioned it.

**Pull request merged**